### PR TITLE
Separate tests into node and browser + airtap + levelup integration tests

### PR DESCRIPTION
**Pull request by ralphtheninja**

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Will need proper setup of sauce labs before adding `test-browsers` script.

File: test-browser.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+var leveljs = require('level-js')
+var testCommon = require('level-js/test/util/test-common')
+var levelup = require('levelup')
+var encoding = require('encoding-down')
+
+require('level-packager/abstract/test')(test, require('./'), {
+  skipRepairTest: true,
+  skipErrorIfExistsTest: true,
+  skipDestroyTest: true
+})
```
Comment (ralphtheninja) [inline test-browser.js]: This code came from `test.js`, but using `level-packager/abstract/test` instead.

File: test-browser.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        db.close(function (err) {
+          t.ifError(err, 'no close error')
+        })
+      })
+    })
+  })
+})
+
+test('teardown', testCommon.tearDown)
```
Comment (ralphtheninja) [inline test-browser.js]: The rest was copied over from `level-js/test/levelup-test.js`

**Comment by vweevers** [comment]
I didn't see where you were going at first. It's coming together now :) Nice work

**Comment by ralphtheninja** [comment]
> I didn't see where you were going at first. It's coming together now :) Nice work

Aye, sorry. Not always easy to convey what's in your head. I've been thinking about this some more and I'll continue on `packager` today and leave this hanging for a bit.

**Comment by ralphtheninja** [comment]
Waiting for a new minor on `level-packager` and then I can finish this off.

**Comment by vweevers** [comment]
@ralphtheninja can you explain the `nonPersistent` option in https://github.com/Level/packager/blob/master/abstract/db-values-test.js?

I must be missing something: if `nonPersistent` defaults to false, where does the data come from?

**Comment by ralphtheninja** [comment]
It's used here https://github.com/Level/mem/blob/master/test.js#L5

**Comment by ralphtheninja** [comment]
> I must be missing something: if nonPersistent defaults to false, where does the data come from?

If the db is persistent (i.e. not memdown), the values are in the db when it's closed. So no need to batch the data. This is what I meant by that the tests are not independent.

**Comment by ralphtheninja** [comment]
@vweevers Are you cool with the latest two commits after your review?

**Pull request merged**